internal/pkg/warewulfd/api/power.go

```diff
@@ -0,0 +1,200 @@
+	"github.com/warewulf/warewulf/internal/pkg/bmc"
+	"github.com/warewulf/warewulf/internal/pkg/node"
+	"github.com/warewulf/warewulf/internal/pkg/wwlog"
+)
+
+// PowerResult is the result structure for single node power status operations. get/set power status
+type PowerResult struct {
+	NodeId string `yaml:"node_id" json:"node_id" description:"ID of the node for which power status is being reported"`
+	BmcIp  string `yaml:"bmc_ip" json:"bmc_ip" description:"BMC IP address of the node"`
+	Result string `yaml:"result" json:"result" description:"Power status of the node, e.g., 'on', 'off', 'unknown'"`
+	Err    error  `yaml:"error,omitempty" json:"error,omitempty" description:"Error message if any error occurred while getting power status"`
+}
+
+// getPower returns an interactor to get the power status of a node by its ID.
+// At this time we ony support a single node. There is no expansion of the hostlist args or batching yet.
+func getPower() usecase.Interactor {
+	type getPowerInput struct {
+		ID string `path:"id" required:"true" description:"ID of node to get power status for"`
+	}
+
+	u := usecase.NewInteractor(func(ctx context.Context, input getPowerInput, output *[]PowerResult) error {
+		wwlog.Debug("api.getPower(), ID: %s", input.ID)
+		powerResults := []PowerResult{}
+		powerResults = append(powerResults, PowerResult{ // start with one expansion is not yet supported
+			NodeId: input.ID,
+		})
+
+		if registry, err := node.New(); err != nil {
+			wwlog.Debug("api.getPower() getting node registry: %v", err)
+			powerResults[0].Err = err // Set the error in the first PowerResult
+			*output = powerResults    // Set *output on exit. important
+			return err
+		} else {
+			if node_, err := registry.GetNode(input.ID); err != nil {
+				wwlog.Debug("api.getPower() node not found: %v", err)
+				powerResults[0].Err = err
+				*output = powerResults // Set *output on exit. important
+				return status.Wrap(fmt.Errorf("node not found: %v (%v)", input.ID, err), status.NotFound)
+			} else {
+				if node_.Ipmi == nil || node_.Ipmi.Ipaddr.IsUnspecified() || node_.Ipmi.UserName == "" || node_.Ipmi.Password == "" {
+					wwlog.Debug("api.getPower() node ipmi not configured")
+					err = status.Wrap(fmt.Errorf("node ipmi not configured"), status.InvalidArgument)
+					powerResults[0].Err = err // Set the error in the first PowerResult
+					*output = powerResults    // Set *output on exit. important
+					return err
+				}
+
+				// starting with powerResults[0] as the accumulator
+				powerResults[0].BmcIp = node_.Ipmi.Ipaddr.String() // Set the BMC IP address in the PowerResult
+
+				// Setup ipmitool call.
+				ipmiCmd := bmc.TemplateStruct{
+					IpmiConf: *node_.Ipmi,
+				}
+
+				// Make ipmitool call.
+				result, err := ipmiCmd.PowerStatus()
+				if err == nil {
+					if result == "Chassis Power is on" {
+						powerResults[0].Result = "on"
+					} else if result == "Chassis Power is off" {
+						powerResults[0].Result = "off"
```

Comment:
These result strings, "Chassis Power is on" and "Chassis Power is off" are, iirc, implementation-specific, and just happen to be what ipmitool returns. That said, I don't know that we _have_ good programmatic differentiation between whether it's "on" or "off" currently.

I suggest, for now, using a regex to see if the result string contains "on". Something like

```
regexp.MustCompile(`(?i)\bon\b`)
```

Then maybe PowerResult could contain the original sring in `Result`, and have a boolean `IsOn` for the parsed value.